### PR TITLE
Update Style+Terminology Guidelines

### PR DESCRIPTION
**Pull request by NHingerl**
**Description**

Changes proposed in this pull request:

- update our [capitalization guidelines](https://kyma-project.io/community/guidelines/content-guidelines/02-style-and-terminology#capitalized-terms) as discussed in SIG Content, July 14
- no article for Kyma components
- switch the order of sections
- arrange terminology in a table
- minor changes

**Related issue(s)**

https://github.com/kyma-project/community/issues/680
https://github.com/kyma-project/kyma/issues/14281
https://github.com/kyma-project/community/issues/481

### CODE REVIEW

**Review by a-thaler — DISMISSED** [review]

File: docs/guidelines/content-guidelines/02-style-and-terminology.md
Comment (majakurcius) [inline docs/guidelines/content-guidelines/02-style-and-terminology.md:63]: While mentioning colons, maybe we could also specify the preferred capitalization after the colon? I remember we discussed it once, whether we should capitalize what comes after the colon, or go with lowercase.

File: docs/guidelines/content-guidelines/02-style-and-terminology.md
Comment (majakurcius) [inline docs/guidelines/content-guidelines/02-style-and-terminology.md:26]: > ⛔️ Click the **Add** button.

Why are we introducing this change? 
Although I'm all for brevity and conciseness, I do believe sometimes the "fuller" version sounds better, especially if using just the name might be ambiguous or confusing, or if there are more UI elements with the same or similar text, or the text from the button is repeated in the sentence.

I'd strongly suggest changing this into a suggestion/recommendation rather than a strict rule and leaving the decision to the writer, who knows the context best.

File: docs/guidelines/content-guidelines/02-style-and-terminology.md
Comment (majakurcius) [inline docs/guidelines/content-guidelines/02-style-and-terminology.md:79]:
```suggestion
* Use the American English spelling, not British English.
```

File: docs/guidelines/content-guidelines/02-style-and-terminology.md
Comment (majakurcius) [inline docs/guidelines/content-guidelines/02-style-and-terminology.md:144]: This is actually not what Upper case means. If you talk about a letter, upper case means that a letter is --simply -- in upper case (`A`, not `a`). If you talk about a piece of text being written in upper case, it means that THE WHOLE TEXT IS WRITTEN IN UPPER CASE JUST LIKE HERE. 
What you mean is probably **Title Case**, where the first letter of each important word is capitalized ("important" meaning that function words, such as 'of', 'and', etc. are excluded from this and written in lowercase).

See the [Cambridge dictionary](https://dictionary.cambridge.org/dictionary/english/upper-case) for reference.

File: docs/guidelines/content-guidelines/02-style-and-terminology.md
Comment (majakurcius) [inline docs/guidelines/content-guidelines/02-style-and-terminology.md:167]:
```suggestion
* CustomResourceDefinition (CRD) - note that "custom resource" alone isn't Kubernetes-specific, thus it's lowercase.
```
In AmE, it's usually written as one word.
See the [Cambridge dictionary](https://dictionary.cambridge.org/dictionary/english/lowercase) for reference.

File: docs/guidelines/content-guidelines/02-style-and-terminology.md
Comment (NHingerl) [reply docs/guidelines/content-guidelines/02-style-and-terminology.md:26]: I agree, if the UI is ambiguous we need to be more specific. But how often does that happen?
99% of the time, people write “Click the **Add** button.” when “Add your XY…” would be perfectly sufficient.

Click-level instructions add dependency on the UI strings, creating maintenance effort to make sure that the UI still says “Add” and didn’t change in the meantime.

File: docs/guidelines/content-guidelines/02-style-and-terminology.md
Comment (NHingerl) [reply docs/guidelines/content-guidelines/02-style-and-terminology.md:144]: Thank you, I'll fix that!

File: docs/guidelines/content-guidelines/02-style-and-terminology.md
Comment (NHingerl) [reply docs/guidelines/content-guidelines/02-style-and-terminology.md:167]: Thank you for spotting this!

File: docs/guidelines/content-guidelines/02-style-and-terminology.md
Comment (NHingerl) [reply docs/guidelines/content-guidelines/02-style-and-terminology.md:29]: How about "Avoid click-level instructions unless the UI is confusing. Readers of Kyma documentation are assumed to be fairly tech-savvy."

File: docs/guidelines/content-guidelines/02-style-and-terminology.md
Comment (NHingerl) [reply docs/guidelines/content-guidelines/02-style-and-terminology.md:70]: I suggest following the [SAP style guide](https://help.sap.com/docs/SNG_FOR_UA/e33c591ae4494a659a3f5f983c9d1161/e2bfb395649b410ba05a213d80092de5.html?locale=en-US&state=DRAFT)?

**Review by a-thaler — APPROVED** [review]

File: docs/guidelines/content-guidelines/02-style-and-terminology.md
Comment (majakurcius) [reply docs/guidelines/content-guidelines/02-style-and-terminology.md:70]: Honestly, I don't mind either. But I wonder if we should not stick to what we already discussed and have in the docs, for example [here](https://kyma-project.io/blog/2021/12/7/release-notes-20/)...

File: docs/guidelines/content-guidelines/02-style-and-terminology.md
Comment (majakurcius) [reply docs/guidelines/content-guidelines/02-style-and-terminology.md]: As a user, I liker (and prefer) click-level instructions. I don't think it's now always about being tech savvy. Imagine if you went to an Azure or AWS dashboard and had no click-level instruction. I'd be lost there, even though the individual controls are self-explanatory. 
And anyway, it's usually the case that the TW writing a document is in charge of maintaining that, so I'd leave it up to the TW. 
If you strongly insist, I'd be OK with adding it as a tip for the undecided that you can decide to follow or not.

Comment (NHingerl) [reply docs/guidelines/content-guidelines/02-style-and-terminology.md]: I don't think we should ever create Azure/AWS instructions, as long as we don't change company? 
If it's our own Kyma software, we hopefully can improve the usability so people don't get lost.

I'm not opposed to documenting navigation paths etc so that users find the correct screen. And of course, instructions must contain every step that's necessary to finish a task successfully. But the level of detail would be different for tech-savvy Kyma users than for, say, the target group of senior phones. 

The [SAP Guide](https://help.sap.com/docs/SNG_FOR_UA/e33c591ae4494a659a3f5f983c9d1161/80cf1399d10b47b482e833b6dbe9211c.html?state=DRAFT&locale=en-US) makes a good point about not stating the obvious; and frankly I think it makes our life easier if the Kyma and SAP style guides don't differ to much.

File: docs/guidelines/content-guidelines/02-style-and-terminology.md
Comment (NHingerl) [reply docs/guidelines/content-guidelines/02-style-and-terminology.md:5]: Good points, thank you!

File: docs/guidelines/content-guidelines/02-style-and-terminology.md
Comment (NHingerl) [reply docs/guidelines/content-guidelines/02-style-and-terminology.md:144]: Much better, thank you!

File: docs/guidelines/content-guidelines/02-style-and-terminology.md
Comment (NHingerl) [reply docs/guidelines/content-guidelines/02-style-and-terminology.md:190]: Let's try it out :) I need to see it rendered, not sure whether it'll overwhelmingly bold or just right.

File: docs/guidelines/content-guidelines/02-style-and-terminology.md
Comment (majakurcius) [reply docs/guidelines/content-guidelines/02-style-and-terminology.md]: > I don't think we should ever create Azure/AWS instructions, as long as we don't change company?
We use external software for development purposes and so sometimes we create documentation that touches that, for example, a guide on how to get logs from the above. 

We hardly ever describe the UI anyway, we do it in the case of Get Started guides, which are targeted for newbies and not only developers, and so being detailed here is justified + it's not like it's huge maintenance if it's just one place. 

Either way, clarity for the user takes priority over our convenience and maintenance load for me. 

I'm still for leaving it up to the author. I think it's a good compromise that we mention it and recommend it but not require it.

Comment (NHingerl) [reply docs/guidelines/content-guidelines/02-style-and-terminology.md]: Well, then especially if it's someone else's GUI, I would not want to provide click-level instructions. It's sad if the Azure/AWS docs aren't good and/or don't provide the instructions that we need, but I'm opposed to closing that gap in Kyma documentation.

Apart from the occasional "from/in the left navigation,", I think the level of detail in the GSG is absolutely fine. Any less detail would make the instructions useless. 

How about "> **TIP:** If possible, avoid..." ? It's softer than "Avoid...!" but not as soft as "Consider avoiding..."

File: docs/guidelines/content-guidelines/02-style-and-terminology.md
Comment (NHingerl) [reply docs/guidelines/content-guidelines/02-style-and-terminology.md:26]:
```suggestion
> **TIP:** If possible, avoid click-level instructions. Readers of Kyma documentation are typically fairly tech-savvy.  
```

File: docs/guidelines/content-guidelines/02-style-and-terminology.md
Comment (grego952) [reply docs/guidelines/content-guidelines/02-style-and-terminology.md]: IMO, we can go with the TIP version, but being responsible for our UI, I must say that the click-level instructions make my life easier, and I believe it is the same for the users. So, tbh I'd stick to the current approach.

Comment (NHingerl) [reply docs/guidelines/content-guidelines/02-style-and-terminology.md]: Then I'd like to go with the "TIP: If possible..." version. Personally, I'm not planning to approve click-level instructions if they can be phrased more abstractly without being less helpful. 

But maybe we're meaning slightly different things here?

Comment (grego952) [reply docs/guidelines/content-guidelines/02-style-and-terminology.md]: For me TIP is fine. Just to make sure we're on the same page, by "click-level" I mean e.g. **Click the Add Button** :)

File: docs/guidelines/content-guidelines/02-style-and-terminology.md
Comment (NHingerl) [inline docs/guidelines/content-guidelines/02-style-and-terminology.md:26]:
```suggestion
> **TIP:** Consider avoiding click-level instructions. Readers of Kyma documentation are typically fairly tech-savvy.  
```

**Review by majakurcius — APPROVED** [review]

**Pull request merged**
